templates/flavors/flatcar/patches/kubeadm-controlplane.yaml

```diff
@@ -0,0 +1,62 @@
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+kind: KubeadmControlPlane
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  kubeadmConfigSpec:
+    diskSetup:
+      filesystems:
+      - device: /dev/disk/azure/scsi1/lun0
+        extraOpts:
+        - -E
+        - lazy_itable_init=1,lazy_journal_init=1
+        filesystem: ext4
+        label: etcd_disk
+      partitions:
+      - device: /dev/disk/azure/scsi1/lun0
+        layout: true
+        overwrite: false
+    format: ignition
+    ignition:
+      containerLinuxConfig:
+        additionalConfig: |
+          systemd:
+            units:
+            - name: kubeadm.service
+              dropins:
+              - name: 10-flatcar.conf
+                contents: |
+                  [Unit]
+                  After=oem-cloudinit.service
+    initConfiguration:
+      nodeRegistration:
+        name: '@@HOSTNAME@@'
+    joinConfiguration:
+      nodeRegistration:
+        name: '@@HOSTNAME@@'
+    mounts:
+    - - etcd_disk
+      - /var/lib/etcddisk
+    postKubeadmCommands: []
+    preKubeadmCommands:
+    - sed -i "s/@@HOSTNAME@@/$(curl -s -H Metadata:true --noproxy '*' 'http://169.254.169.254/metadata/instance?api-version=2020-09-01'
+      | jq -r .compute.name)/g" /etc/kubeadm.yml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  template:
+    spec:
+      image:
+        sharedGallery:
+          gallery: ClusterAPI
+          name: capi-flatcar-stable-2905.2.3
+          offer: flatcar-container-linux-free
+          publisher: kinvolk
+          resourceGroup: cluster-api-images
+          sku: stable
+          subscriptionID: ${AZURE_SUBSCRIPTION_ID}
+          version: 0.3.1632385693
```

Comment:
Does this require the user to build their own image in a SIG first? if so, can we put some docs around it? Also, where is this version coming from?

Reply:
The version contains the build timestamp and comes from here: https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/packer/azure/azure-sig.json.

At this time: yes, the user has to build their own images. There is existing documentation on building SIG images here https://capz.sigs.k8s.io/topics/custom-images.html#using-shared-image-gallery-recommended, and it even mentions Flatcar already :) What extensions to the documentation would you suggest?

Moving forward I'd love to see new marketplace images published for testing/demo purposes. I'd love to hear your thoughts on that. We could either publish third-party images under the existing "Kinvolk" publisher, or perhaps first-party is an option?

Reply:
I think I found a better way:
```
version: latest
```
is also correct.